File: 0070.Climbing_Stairs.js
//0ms

/**
 * @param {number} n
 * @return {number}
 */
var climbStairs = function(n) {
    let secondLast = 0;
    let last = 1;
    for (let i = 1; i <= n; i++) {
        let temp = last;
        last = secondLast + last;
        secondLast = temp;
    }
    return last;
};

File: 0070.Climbing_Stairs.rs
//0ms


impl Solution {
    pub fn climb_stairs(n: i32) -> i32 {
        let mut second_last = 0;
        let mut last = 1;
        for _ in 1..=n {
            let temp = last;
            last = second_last + last;
            second_last = temp;
        }
        last
    }
}
